First eccentricity calculation

NGoetz opened the pull request:
This PR adds eccentricity calculation in two ways (particle and grid based), and joins the EventCharacteristics and Lattice3D extensions.

Hendrik1704 commented on src/particle-analysis/EventCharacteristics.py:
```diff
@@ -0,0 +1,81 @@
+        for particle in particle_data:
+            x = particle.x
+            y = particle.y
+            rn = x**harmonic_n+y**harmonic_n
```
Shouldn't this be `rn = (x**2. + y**2.)**(harmonic_n/2.)`, because of $r=\sqrt{x^2+y^2}$ and there should be an if statement if someone computes $\varepsilon_1$. This is usually computed with a $r^3$ weight instead of $r$.

NGoetz replied:
Indeed you are right. I was not aware of the special case for dipole asymmetry.

Hendrik1704 commented on src/particle-analysis/EventCharacteristics.py:
```diff
@@ -0,0 +1,81 @@
+            real_eps += rn*np.cos(harmonic_n*t)*particle.E
+            imag_eps += rn*np.sin(harmonic_n*t)*particle.E
+            norm += rn*particle.E
+        return np.mean(real_eps)/np.mean(norm)+np.mean(imag_eps)/np.mean(norm)*1j
```
Why the mean function for a single number?

Hendrik1704 commented on src/particle-analysis/EventCharacteristics.py:
```diff
@@ -0,0 +1,81 @@
+            y = particle.y
+            rn = x**harmonic_n+y**harmonic_n
+            t = np.arctan2(y,x)
+            real_eps += rn*np.cos(harmonic_n*t)*particle.E
```
For the particles it would be possible to have the option to weight the eccentricities by the particles (`particles.E`  $\rightarrow$ `1`) and not by the energy. This could be an option in the function call.

NGoetz replied:
I think when we are at it, we should also allow charge and baryon number weighting. It is not very common but there are usecases.

Hendrik1704 commented on src/particle-analysis/EventCharacteristics.py:
```diff
@@ -0,0 +1,81 @@
+            norm += rn*particle.E
+        return np.mean(real_eps)/np.mean(norm)+np.mean(imag_eps)/np.mean(norm)*1j
+    
+    def eccentricity_from_energy(self,harmonic_n,lattice):
```
Same comments as for the function above.

Hendrik1704 commented on src/particle-analysis/Lattice3D.py:
```diff
@@ -0,0 +1,364 @@
+        self.num_points_x_ = num_points_x
+        self.num_points_y_ = num_points_y
+        self.num_points_z_ = num_points_z
+        self.cell_volume = (x_max-x_min)*(y_max-y_min)*(z_max-z_min)/(num_points_x*num_points_y*num_points_z)
```
Stick to the same notation as for the other quantities: `self.cell_volume_`

Hendrik1704 reviewed (changes requested):
I'm not convinced by the formula you implemented for the eccentricity. To me it looks wrong.
The rest is mostly fine for me.

NGoetz commented:
I have adressed the mentioned issues and generalized the definition of both eccentricies. I also added some refactoring and introduced an option to reset the lattice.
